### PR TITLE
Add xsuaa certificate authentication

### PR DESCRIPTION
**Pull request by sbabev**

### CODE REVIEW

File: src/authentication.ts [hunk 1]
```diff
@@ -168,4 +212,9 @@ export class OAuthAuthentication implements Authentication {
                 });
         });
     }
+
+    // eslint-disable-next-line require-jsdoc
+    private isCertificateAuthentication(config: OAuthConfig): boolean {
+        return !config.password && config.certificate != null && config.privateKey != null;
```
Comment (mkunev) [inline src/authentication.ts:218]: Should be changed to the following:
```suggestion
        return !config.password && !!config.certificate && !!config.privateKey;
```
The current behaviour will return true in case of empty string for certificate/privateKey.
Add some tests to verify what will happen if certificate/privateKey is empty or not passed (maybe add some verification that they are not empty strings?)

File: docs/_data/oauth-authentication.json [hunk 2]
```diff
@@ -10,6 +10,16 @@
             "type": "string",
             "description": "A technical client's password"
         },
+        {
+            "name": "certificate",
+            "type": "string",
+            "description": "x509 certificate chain generated by xsuaa service"
```
Comment (yusmenrashidov) [inline docs/_data/oauth-authentication.json:16]: UAA service or something else

File: docs/_data/oauth-authentication.json [hunk 3]
```diff
@@ -10,6 +10,16 @@
...
+        },
+        {
+            "name": "privateKey",
+            "type": "string",
+            "description": "Private key related to the generated certificate"
```
Comment (yusmenrashidov) [inline docs/_data/oauth-authentication.json:21]: Private key of the generated certificate

Comment (sbabev) [reply docs/_data/oauth-authentication.json:21]: I suggest to let it be like this because in the documentation for mTLS it is written like here.

**Review by mkunev — APPROVED** [review]
